deploy/charts/jetstack-agent/README.md

```diff
@@ -137,19 +137,24 @@ kubectl logs -n jetstack-secure $(kubectl get pod -n jetstack-secure -l app.kube
 | authentication.secretKey | string | `"credentials.json"` | Key name in secret |
 | authentication.secretName | string | `"agent-credentials"` | Name of the secret containing agent credentials.json |
 | authentication.secretValue | string | `""` | Base64 encoded value from Jetstack Secure Dashboard - only required when createSecret is true |
-| authentication.type | string | `"file"` | Type can be "file"/"token" which dictates whether a credential file or token is used |
-| config | object | `{"cluster":"","dataGatherers":{"custom":[],"default":true},"organisation":"","period":"0h1m0s","server":"https://platform.jetstack.io"}` | Configuration section for the Jetstack Agent itself |
+| authentication.type | string | `"file"` | Type can be "file"/"token" determining how the agent should authenticate the to the backend |
+| command | list | `[]` |  |
+| config | object | `{"cluster":"","dataGatherers":{"custom":[],"default":true},"organisation":"","override":{"config":"","configmap":{"key":"","name":""},"enabled":false},"period":"0h1m0s","server":"https://platform.jetstack.io"}` | Configuration section for the Jetstack Agent itself |
 | config.cluster | string | `""` | REQUIRED - Your Jetstack Secure Cluster Name |
 | config.dataGatherers | object | `{"custom":[],"default":true}` | Configure data that is gathered from your cluster, for full details see https://platform.jetstack.io/documentation/configuration/jetstack-agent/configuration |
 | config.dataGatherers.custom | list | `[]` | A list of data gatherers to limit agent scope |
 | config.dataGatherers.default | bool | `true` | Use the standard full set of data gatherers |
 | config.organisation | string | `""` | REQUIRED - Your Jetstack Secure Organisation Name |
+| config.override | object | `{"config":"","configmap":{"key":"","name":""},"enabled":false}` | Provide an Override to allow completely custom agent configuration |
+| config.override.config | string | `""` | Embed the agent configuration here in the chart values |
+| config.override.configmap | object | `{"key":"","name":""}` | Sepcify ConfigMap details to load config from existing ConfigMap |
+| config.override.enabled | bool | `false` | Override disabled by default |
```

Comment:
Could we make this more implicit? like set the config.override to null and if it has set config or configmap it is automatically "enabled"? just a small UX preference for me, i like to use enabled but for bigger pieces of config if that makes sense..

Reply:
We explored the option and found it's just too dirty in helm templating to make this work nicely.